### PR TITLE
Made additional retryable errors

### PR DESCRIPTION
**Pull request by joereuss12**
Made StoppedTransferError and Cannot allocate memory a retryable error

### CODE REVIEW

**Review by haoming29 — APPROVED** [review]
LGTM!

File: client/errorAccum.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func (te *TransferErrors) AddPastError(err error, timestamp time.Time) {
 	}
 }
 
+// This resets the TransferErrors object for testing purposes
+func (te *TransferErrors) ResetErrors() {
```
Comment (bbockelm) [inline client/errorAccum.go:75]: It seems this is only used by unit tests - please change to not be exported outside the module.

File: client/handle_http.go [hunk 2]
```diff
@@ -112,6 +112,10 @@ type (
 		Err error
 	}
 
+	AllocateMemoryError struct {
```
Comment (bbockelm) [inline client/handle_http.go:115]: It doesn't seem this needs to be visible outside the module - please don't export.

**Review by bbockelm — CHANGES_REQUESTED** [review]
Basic idea is good, just nits about the visibility.  Since we want others to eventually use the client module, we need to be extremely careful about exported types and methods.

**Review by bbockelm — APPROVED** [review]

**Pull request merged**